### PR TITLE
feat(node): add feature flag for tcp/quic

### PR DESCRIPTION
**Pull request by aed900**
Added a feature flag to be able to toggle quic instead of the tcp default.

### CODE REVIEW

File: sn_networking/src/lib.rs [hunk 1]
```diff
@@ -192,7 +196,20 @@ impl SwarmDriver {
...
+        let _listener_id = swarm_driver
+            .swarm
+            .listen_on(addr)
+            .expect("Failed to listen on the provided address");
+
+        #[cfg(feature = "quic")]
+        let addr = Multiaddr::from(addr.ip())
+            .with(Protocol::Udp(addr.port()))
+            .with(Protocol::QuicV1);
+        #[cfg(feature = "quic")]
```
Comment (b-zee) [inline sn_networking/src/lib.rs:212]:
```suggestion
        #[cfg(not(feature = "quic"))]
        let addr = Multiaddr::from(addr.ip())
            .with(Protocol::Tcp(addr.port()));
        #[cfg(feature = "quic")]
        let addr = Multiaddr::from(addr.ip())
            .with(Protocol::Udp(addr.port()))
            .with(Protocol::QuicV1);
```

File: sn_networking/src/lib.rs [hunk 2]
```diff
@@ -356,9 +373,16 @@ impl SwarmDriver {
             .multiplex(libp2p::yamux::Config::default())
             .boxed();
 
+        #[cfg(feature = "quic")]
+        let quic_transport = libp2p_quic::tokio::Transport::new(quic::Config::new(&keypair));
+        #[cfg(feature = "quic")]
+        let mut transport = quic_transport
+            .map(|(peer_id, muxer), _| (peer_id, StreamMuxerBox::new(muxer)))
+            .boxed();
```
Comment (RolandSherwin) [inline sn_networking/src/lib.rs:381]: Just a nit pick, we can probably combine these for better readability?
```suggestion
        #[cfg(feature = "quic")]
        let mut transport = libp2p_quic::tokio::Transport::new(quic::Config::new(&keypair))
            .map(|(peer_id, muxer), _| (peer_id, StreamMuxerBox::new(muxer)))
            .boxed();
```

**Review by RolandSherwin — APPROVED** [review]
PR looks good to me!

File: sn_networking/src/lib.rs [hunk 3]
```diff
@@ -356,9 +373,16 @@ impl SwarmDriver {
...
+
```
Comment (RolandSherwin) [inline sn_networking/src/lib.rs]: Just a nit pick, but can we probably combine these for better readability?
```suggestion
        #[cfg(feature = "quic")]
        let mut transport = libp2p_quic::tokio::Transport::new(quic::Config::new(&keypair))
            .map(|(peer_id, muxer), _| (peer_id, StreamMuxerBox::new(muxer)))
            .boxed();

```

File: sn_networking/src/lib.rs [hunk 4]
```diff
@@ -196,7 +200,19 @@ impl SwarmDriver {
...
+            .expect("Failed to listen on the provided address");
+
+        #[cfg(feature = "quic")]
+        let addr = Multiaddr::from(addr.ip())
+            .with(Protocol::Udp(addr.port()))
+            .with(Protocol::QuicV1);
+        #[cfg(feature = "quic")]
         let _listener_id = swarm_driver
             .swarm
             .listen_on(addr)
```
Comment (RolandSherwin) [inline sn_networking/src/lib.rs]: Another nit pick, we can remove the cfg for the `listen_on` call here?
```suggestion
        #[cfg(not(feature = "quic"))]
        let addr = Multiaddr::from(addr.ip()).with(Protocol::Tcp(addr.port()));

        #[cfg(feature = "quic")]
        let addr = Multiaddr::from(addr.ip())
            .with(Protocol::Udp(addr.port()))
            .with(Protocol::QuicV1);
        let _listener_id = swarm_driver
            .swarm
            .listen_on(addr)
            .expect("Failed to listen on the provided address");
```

**Comment by joshuef** [comment]
Nice one, thanks @aed900 ! 🙇

**Pull request merged**